flake.nix

```
@@ -1,206 +1,106 @@
-          nativeBuildInputs = [
-              (ghcVersion pkgs).cabal-install
-              pkgs.hpack
-              pkgs.jq
-              pkgs.nix
-              pkgs.z3
-              pkgs.lsof
-          ];
-          shellHook = ''
-            hpack booster && hpack dev-tools
-          '';
+
+          devShells.default =
+            let haskellPackages = pkgs.haskell.packages.${ghcVer};
+            in
+            haskellPackages.shellFor {
+              packages = p: [ self.packages.${system}.sample ];
+              withHoogle = false;
+              buildInputs = with haskellPackages; [
+                cabal-install
+              ];
+            };
         };
-      });
+    in
+    flake-utils.lib.eachDefaultSystem out // {
+      overlays = {
+        haskell-versions = makeHaskellOverlay (final: prev:
+          import ./nix/some-cabal-hashes.nix {
+            self = final;
 
-      devShell =
-        perSystem (system: (nixpkgsFor system).haskell-backend.devShell);
+            overrides = 
+              let 
+                hprev = prev.haskell.packages;
+                hfinal = final.haskell.packages;
+              in {
+                # FIXME these two overrides cause an error in some-cabal-hashes.nix
+                # tasty-test-reporter = 
+                #     final.fetchFromGitHub {
+                #       owner = "goodlyrottenapple";
+                #       repo = "tasty-test-reporter";
+                #       rev = "b704130545aa3925a8487bd3e92f1dd5ce0512e2";
+                #       sha256 = "sha256-9CN59Im0BC3vSVhL85v5eXPYYoPbV3NAuv893tXpr/U=";
+                #     };
+                # profiteur =
+                #     final.fetchFromGitHub {
+                #       owner = "goodlyrottenapple";
+                #       repo = "profiteur";
+                #       rev = "7b30bbe6b2a6b72a5b4896f3a0eed5587a5bf465";
+                #       sha256 = "sha256-9CN59Im0BC3vSVhL85v5eXPYYoPbV3NAuv893tXpr/U=";
+                #     };
 
-      overlays = {
-        z3 = (final: prev: {
-          z3 = prev.z3.overrideAttrs (_: {
-            src = z3;
-            version = let
-              release = builtins.readFile "${z3}/scripts/release.yml";
-              # Read the release version from scripts/release.yml
-            in builtins.head
-            (builtins.match ".+ReleaseVersion: '([^']+).+" release);
+                # FIXME a jailbreak does not work either (Problem is the ansi-terminal version)
+                # hfinal.tasty-test-reporter = hfinal.doJailbreak hprev.tasty-test-reporter;
+
+                # FIXME likewise, these two `dontCheck` (one inlined) are seemingly ignored.
+                hfinal.decision-diagrams = hfinal.overrideCabal hprev.decision-diagrams 
+                    (drv: { doCheck = false; }); # build runs failing tests
+                hfinal.json-rpc = final.haskell.packages.dontCheck hprev.json-rpc; # 1.0.4 marked as broken!
```

This `dontCheck` setting is somehow also ignored, I keep getting a build failure because of a failing test in `decision-diagrams`

---

This should be due to the attribute names that are being overridden, here. In this override example, you have to use the haskell package names directly, if I understand correctly. Therefore, currently, the original variants of the packages are being used instead.

```suggestion
                decision-diagrams = hfinal.overrideCabal hprev.decision-diagrams 
                    (drv: { doCheck = false; });
                json-rpc = hfinal.dontCheck hprev.json-rpc;
```

---

I tried this earlier, but `nix build` (`--dry-run` or not) complained about the `hfinal.overrideCabal` (or `hfinal.dontCheck`) as a missing attribute. Changing that to `prev.haskell.lib.{overrideCabal,dontCheck}` worked better :smirk: yet still the build is failing on `decision-diagrams` (but differently now)